snippets/base/models.py

```diff
@@ -668,10 +668,46 @@ def subtemplate(self):
                 except Template.DoesNotExist:
                     continue
 
+    def _convert_special_buttons(self, data):
+        to_delete = []
+        to_append = {}
+        for key, value in data.items():
+            if key == 'button_url':
+                if value == 'special:appMenu':
+                    to_append['button_action'] = 'OPEN_APPLICATIONS_MENU'
+                    to_append['button_action_args'] = 'appMenu'
+                    to_delete.append(key)
+                elif value == 'special:accounts':
+                    to_append['button_action'] = 'SHOW_FIREFOX_ACCOUNTS'
+                    to_delete.append(key)
+                elif value == 'special:monitor':
+                    to_append['button_action'] = 'ENABLE_FIREFOX_MONITOR'
+                    to_append['button_action_args'] = {
+                        'url': ('https://monitor.firefox.com/oauth/init?'
+                                'utm_source=desktop-snippet&utm_term=[[snippet_id]]&'
+                                'utm_content=[[channels]]&utm_campaign=[[campaign_slug]]&'
+                                'entrypoint=snippets&form_type=email'),
+                        'flowRequestParams': {
+                            'entrypoint': 'snippets',
+                            'utm_term': 'monitor',
+                            'form_type': 'email'
+                        }
+                    }
+                    to_delete.append(key)
+
+        for key in to_delete:
+            data.pop(key)
+
+        data.update(to_append)
+
+        return data
```

Comment:
There's not really a need to return `data` since you're modifying it in place. I'd suggest just for clarity that you either copy `data` at the start of this function and return the new modified object, or just call this on `data` below and rely on it being modified in place. Not required as this also works, but returning and assigning to `data` below on line 709 is really a noop.

Reply:
Fair point. I like to return data from my functions, so I'll go ahead and create a local copy  as you suggest. Thanks for the better code recommendation!